src/wp-includes/class-http.php

```diff
@@ -305,6 +305,12 @@ public function request( $url, $args = array() ) {
 			$options['hooks']->register( 'requests.before_redirect', array( get_class(), 'validate_redirects' ) );
 		}
 
+		// Allow overriding the protocol version.  The 'httpversion' argument
+		// claims to do this but appears to be broken.
+		if ( isset( $r['protocol_version'] ) ) {
+			$options['protocol_version'] = $r['protocol_version'];
+		}
+
```

Comment:
Should we instead fix any problem in the `httpversion` argument?

Reply:
There are possible compatibility issues if we fix something that has been broken for a long time.

Reply:
Consider changing this option name to `cp_http_protocol_version` instead for improved compatibility? Need to search plugins/themes for existing uses of `protocol_version`